client/client.go

```diff
@@ -316,52 +309,56 @@ type Config struct {
 	NoVerify   bool
 }
 
-func loadServerTrust(conf Config) (*x509.CertPool, error) {
-	if conf.ServerCert == "" {
-		// Returning nil will make tls.Config.RootCAs nil, which causes
-		// tls module to use system certs.
-		return nil, nil
-	}
+type systemCertPoolGetter interface {
+	GetSystemCertPool() (*x509.CertPool, error)
+}
+
+type systemCertPool struct{}
+
+func (systemCertPool) GetSystemCertPool() (*x509.CertPool, error) {
```

Comment:
Suggestion: This is an initializer right?
How about `systemCertPool.New()` instead?

Reply:
No, this is the actual loading code. This calls `x509.SystemCertPool()`, the initializer is `x509.NewCertPool()` (which we don't expose, since we don't need it). For this reason we should avoid the name `New()`.

Reply:
Alright, I misunderstood. Thank you for the explanation.